plugins/ingest/src/test/java/org/elasticsearch/ingest/processor/grok/GrokProcessorTests.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest.processor.grok;
+
+import org.elasticsearch.ingest.IngestDocument;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.*;
+
+import static org.hamcrest.Matchers.*;
+
+
+public class GrokProcessorTests extends ESTestCase {
+
+    public void testMatch() throws Exception {
+        Map<String, Object> source = new HashMap<>();
+        source.put("foo", "1");
```

Comment:
I think he means to do something like this

```java
String fieldName = RandomDocumentPicks.randomFieldName(random());
IngestDocument doc = new IngestDocument("index", "type", "id", new HashMap<>());
doc.setFieldValue(fieldName, "1");
```

this way, passing a string "foo.bar" would appropriately be treated as an object path in the context of `IngestDocument#setFieldValue`, rather than simply doing a `put` on a normal Map.